contrib/check_model.py

```diff
@@ -0,0 +1,131 @@
+import sys
+from collections import defaultdict
+from followthemoney import model
+
+
+IGNORE_DIVERGENT_TYPES = [
+    "author",
+    "organization",
+    "gender",
+    "number",
+    "authority",
+    "duration",
+    "area",
+    "subject",
+    "sender",
+]
+
+IGNORE_DIVERGENT_LABELS = [
+    "parent",
+    "holder",
+    "number",
+    "authority",
+    "criteria",
+    "procedure",
+]
+
+IGNORE_LABEL_COLLISIONS = [
+    "Address",
+    "Notes",
+    "Customs declarations",
+    "Country of origin",
+    "Payments received",
+    "Payments made",
+    "Entity",
+    "The language of the translated text",
+    "Responding to",
+    "ISIN",
+    "Document number",
+]
+
+
+def test_divergent_types(by_name):
+    divergent = {}
+
+    for name, props in by_name.items():
+        if len(props) == 1 or name in IGNORE_DIVERGENT_TYPES:
+            continue
+
+        types = set([p.type for p in props])
+        if len(types) > 1:
+            divergent[name] = props
+
+    return divergent
+
+
+def test_divergent_labels(by_name):
+    divergent = {}
+
+    for name, props in by_name.items():
+        if len(props) == 1 or name in IGNORE_DIVERGENT_LABELS:
+            continue
+        
+        labels = set([p.label for p in props])
+        if len(labels) > 1:
+            divergent[name] = props
+
+    return divergent
+
+
+def test_label_collisions(by_label):
+    collisions = {}
+
+    for label, props in by_label.items():
+        if len(props) == 1 or label in IGNORE_LABEL_COLLISIONS:
+            continue
+
+        names = set([p.name for p in props])
+        if len(names) > 1:
+            collisions[label] = props
+
+    return collisions
+
```

Comment:
While they use the same structure, they do different things, and abstracting the generic structure would probably require passing a bunch of parameters or predicates as lambdas. I’m not sure this would make it easier to understand/maintain tbh